ios/RCTMapboxGL/RCTMapboxAnnotation.m

```diff
@@ -17,4 +17,14 @@
 @implementation RCTMapboxAnnotation {
 }
 
+-(NSString *)reuseIdentifier {
+    return self.id;
+}
+
+-(void)layoutSubviews {
+    [super layoutSubviews];
+    [self.map restoreAnnotationPosition:self.id];
```

Comment:
If the intent is to maintain the view’s center after changing its size, you can instead:

1. Store the current `center` in a local variable.
1. Call `[super layoutSubviews]`.
1. Set the `center` back to the value stored in (1).